example_client/grpc_stateful_client.py

```diff
@@ -32,154 +32,250 @@ def printDebug(msg):
+            request = predict_pb2.PredictRequest()
+            request.model_spec.name = args.get('model_name')
+
+            # Set input data index
+            input_index = x
+            # Input for context window
+            if x < cw_l:
+                # Left context window
+                input_index = 0
+            elif x >= cw_l and x < batch_size + cw_l:
+                # Middle window
+                input_index = x-cw_l
+            else:
+                # Right context window
+                input_index = batch_size - 1
+
+            # Setting request input
+            for input_name in input_names:
+                inputSubArray = inputArrays[input_name]
+                inputData = inputSubArray[key][input_index]
+                request.inputs[input_name].CopyFrom(make_tensor_proto(inputData, shape=inputData.shape))
+
+            # Add sequence start
+            if x == 0:
+                request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
+        
+            # Set sequence id
+            request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
 
-    samples_limit = int(args.get('samples'))
-    for x in range(0, batch_size):
+            # Add sequence end
+            if x == batch_size + cw_l + cw_r - 1:
+                request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_END, dtype="uint32"))
 
-        if x >= samples_limit:
-            break
+            start_time = datetime.datetime.now()
+            result = stub.Predict(request, 10.0) # result includes a dictionary with all model outputs
+            end_time = datetime.datetime.now()
 
-        printDebug('\tExecution: {}\n'.format(x))
-        request = predict_pb2.PredictRequest()
-        request.model_spec.name = args.get('model_name')
+            ValidateOutput(result, output_names)
 
-        printDebug('\tTensor before input in shape: {}\n'.format(obj[x].shape))
-        inputArray = np.expand_dims(obj[x], axis=0)
-        printDebug('\tTensor input in shape: {}\n'.format(inputArray.shape))
+            duration = (end_time - start_time).total_seconds() * 1000
+            processing_times = np.append(processing_times,np.array([int(duration)]))
 
-        request.inputs[args['input_name']].CopyFrom(make_tensor_proto(inputArray, shape=inputArray.shape))
+            # Compare results after we are pass initial context window results
+            if score_index >= 0:
+                # Loop over reference output results
+                errPerNameSum = 0.0
 
-        if x == 0:
-            request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
-        
-        request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
+                for output_name in output_names:
+                    scoreData = referenceArrays[output_name][key][score_index]
```

Comment:
Refering to `key` here assumes that all input and score arks' have utterances with exactly the same names.